src/compiler/transformers/declarations.ts

```diff
@@ -214,7 +215,16 @@ import {
 /** @internal */
 export function getDeclarationDiagnostics(host: EmitHost, resolver: EmitResolver, file: SourceFile | undefined): DiagnosticWithLocation[] | undefined {
     const compilerOptions = host.getCompilerOptions();
-    const result = transformNodes(resolver, host, factory, compilerOptions, file ? [file] : filter(host.getSourceFiles(), isSourceFileNotJson), [transformDeclarations], /*allowDtsFiles*/ false);
+    const files = filter(getSourceFilesToEmit(host, file), isSourceFileNotJson);

```

Comment:
@weswigham i am not sure if i should have just file here in the transform and it is ok or do you need to have all the files and then filter results from only that file ? (this is because of --out when all files need to be emitted together in a bundle) we have a test that failed without this change `getDeclarations` in fourslash

Reply:
`getDeclarationDiagnostics` already differs slightly from normal declaration emit's transform call when `--out` is supplied, so I wouldn't sweat it - the little differences historically haven't mattered, since declaration emit-added errors are all visibility related (plus isolatedDeclaration related now, which are all very context-free), and generally don't depend on the input file set. Specifically, `emitFiles` will put all the files into a `Bundle` when `--out` is specified, but this does not. That difference has yet to matter in practice, since we don't have any bundle-specific errors (that are not instead switched based on compiler options), so I'm pretty sure just sending in the interesting file even when `--out` is set is probably OK.